Yosemite/Yosemite/Stores/DomainStore.swift

```diff
@@ -39,8 +39,12 @@ public final class DomainStore: Store {
 private extension DomainStore {
     func loadFreeDomainSuggestions(query: String, completion: @escaping (Result<[FreeDomainSuggestion], Error>) -> Void) {
         Task { @MainActor in
-            let result = await remote.loadFreeDomainSuggestions(query: query)
-            completion(result)
+            do {
```

Comment:
I was going to suggest `Result(catching:)` here but then realized it's async. In the original conversation, I suggested an extension to make an async catching initializer:

```swift
extension Result where Failure == Error {
    init(catching body: () async throws -> Success) async {
        do {
            let value = try await body()
            self = .success(value)
        } catch {
            self = .failure(error)
        }
    }
}
```

With that, this could be simplified as:

```swift
private extension DomainStore {
    func loadFreeDomainSuggestions(query: String, completion: @escaping (Result<[FreeDomainSuggestion], Error>) -> Void) {
        Task { @MainActor in
            let result = await Result {
                try await remote.loadFreeDomainSuggestions(query: query)
            }
            completion(result)
        }
    }
}
```

I see this also apply to several other diffs in Yosemite, so it might make sense to add it on this PR

Reply:
Nice improvement! Added in b00cd6aa3b8590bee3b98455ed680b2385f12c10